src/text_snippets.cpp

```diff
@@ -43,9 +43,9 @@ void snippet_library::add_snippet_from_json( const std::string &category, JsonOb
 
 int snippet_library::add_snippet(const std::string &category, const std::string &text)
 {
-    int hash = djb2_hash( (const unsigned char *)text.c_str() );
-    snippets.insert( std::pair<int, std::string>(hash, text) );
-    categories.insert( std::pair<std::string, int>(category, hash) );
+    auto const hash = static_cast<int>(std::hash<std::string> {}(text));
+    snippets.insert( make_pair(hash, text) );
+    categories.insert( make_pair(category, hash) );
```

Comment:
@bkentel Just as a note: the hash of the snippet is stored in the saves along the item data, so a player always sees the same snippet when they look at the item description. Changing the hash function will have a negative effect on save  compatibility.